### DISCUSSION TITLE
Potential support for Seeed Studio XIAO ESP32C3 ?

### DISCUSSION THREAD

**Discussion by stamandster** [mp-se/gravitymon#166] [Ideas]
I'd love to be able to use the following Seeed Studio XIAO ESP32C3 board, however, the pins aren't the same as the other supported boards :/ I just don't seem to have the mental fortitude to get the PlatformIO configured and working to try to add it myself (more of a normal Arduino IDE guy). 

The size of this board allows for a greater variation of containers, along with not needing an additional charging daughter board. It also has an external antenna that can be changed out and/or attached to the inside for potentially better wifi/ble coverage.

Preferably the following PINS

PIN_VOLT - Power Monitor
GPIO2 - D0/A0

MPU6050 Gyro
PIN_SDA -> GPIO6 - D4 - SDA
PIN_SDA -> GPIO7 - D5 - SCL

PIN_DS - Temp Sensor DS18B20
GPIO9 - D9

PIN_LED - Built-in LED
GPIO10 - D10

https://docs.platformio.org/en/latest//boards/espressif32/seeed_xiao_esp32c3.html

[env:seeed_xiao_esp32c3]
platform = espressif32
board = seeed_xiao_esp32c3

; change MCU frequency
board_build.f_cpu = 160000000L

https://wiki.seeedstudio.com/XIAO_ESP32C3_Getting_Started/
https://forum.seeedstudio.com/t/battery-voltage-monitor-and-ad-conversion-for-xiao-esp32c/267535

**Comment by mp-se** [comment]
Since its a c3 chip it would be possible to add support for this, but this will require a whole new hardware setup and a lot of testing to ensure it works as intended. I dont have any hardware that I can try this on but I can give you tips on what needs to be changed. I would suggest that you get accustomed to using platformio first. 

I have had issues with the standard c3 chips, especially related to wifi connections which drains a lot of power. 

The standard chips fit the current hardware design quite nice, im planning to support the new s3 mini board as the next step

**Reply by stamandster** [reply]
Ok, that sounds like a fair compromise and I appreciate the help :)  My main issue is there are things that are broken for me out of the gate. Not sure what dependencies I need to add to resolve. I also signed up for and configured my platformio account. I installed the Wokwi Simulator but still have the errors below.

platformio.ini
Ignore unknown configuration option `gcov_files` in section [env:cov-wokwi]
Ignore unknown configuration option `gcov_files` in section [env:cov-esp32]

Console ->
Warning! Ignore unknown configuration option `gcov_files` in section [env:cov-wokwi]
Warning! Ignore unknown configuration option `gcov_files` in section [env:cov-esp32]
Warning! Ignore unknown configuration option `gcov_files` in section [env:cov-wokwi]
Warning! Ignore unknown configuration option `gcov_files` in section [env:cov-esp32]
Resolving gravity-release dependencies...
MissingPackageManifestError: Could not find one of 'package.json' manifest files in the package

**Comment by mp-se** [comment]
You can just ignore those warnings, the gcov_files is a custom tag that I use when testing for code coverage. comment them out if you want to get rid of the warning.  The "MissingPackageManifestError" is something I havent seen before, dont know why that is shown.

**Comment by stamandster** [comment]
Thanks! I'm working through the error also on my side with clearing cache and packages, wonder if there's some sort of corruption

**Comment by mp-se** [comment]
Most of the changes you need to do is in the main.hpp file. 

0) create a new target in the platformio.ini file for the device

1) Create a new define that maps toward the hardware device, i use ESP32C3 for the standard c3 mini so create a new define such as XIAO_ESP32C3 or similar.

2) Create a new section in main.hpp that defines all the needed pins. you will need SDA,SCL,DS,VOLT as the standard pins. CFG1/2 are there to force the device into config mode if they are connected. You will also need to map RX/TX pins to expose the internal serial port which is needed to debug startup on the chips with USBC.

3) Finally you need to fix the serial console output to steer that to the TX/RX pins when the USE_SERIAL_PINS is enabled. This is in the helper.cpp file. 

That should be the main hardware things to sort out. Then its just debugging and validating that its stable. Some problems can be hard to find without a proper debugger attachesd

**Reply by stamandster** [reply]
Awesome! Thanks for the direction on trying to get this tested out and working :) I'll do my best in the next week or so and report back! Thank you again!
